koku/koku/celery.py

```diff
@@ -127,6 +127,18 @@ def validate_cron_expression(expresssion, default="0 * * * *"):
     }
     app.conf.beat_schedule["check-report-updates-batched"] = CHECK_REPORT_UPDATES_DEF
 
+    # The schedule to scan for XL reports
+    xl_download_fallback = validate_cron_expression("0 */3 * * *")
+    xl_download_expression = ENVIRONMENT.get_value("XL_REPORT_DOWNLOAD_SCHEDULE", default=xl_download_fallback)
+    XL_REPORT_DOWNLOAD_SCHEDULE = validate_cron_expression(xl_download_expression, xl_download_fallback)
+    xl_report_schedule = crontab(*XL_REPORT_DOWNLOAD_SCHEDULE.split(" ", 5))
+    CHECK_REPORT_UPDATES_XL = {
+        "task": download_task,
+        "schedule": xl_report_schedule,
+        "kwargs": {"large_providers": "True"},
+    }
+    app.conf.beat_schedule["check-report-updates-batched"] = CHECK_REPORT_UPDATES_XL
```

Comment:
I think we are going to get very confused trying to tweak this schedule, the batch size, and the report count. 

Should we just run this hourly instead?

I also feel like this runs the risk of polling 10 large providers at a time which might defeat the purpose of what we are trying to do.

Reply:
Confusion on configuring the schedule is certainly a good point. I was taking a count of the largest providers and trying to split those specifically through the day. That count today is 8 so thats why I landed at 3 by default. This could be configured to be more frequent though. 

As for triggering it hourly, we can but the idea is to spread the load through the day as much as possible because this large providers take between 1-5 hours to process. So with 3 hours hopefully gives us a little more time to process before the next provider starts.

10 large providers risk is exactly what we have today with the current polling, their is certainly a chance that all the large providers could be collected within our 25 batch size. This new strategy would be controlled by the the XL_POLLING_BATCH_SIZE. The default being 1 right now.